sycl/tools/sycl-trace/collector.cpp

```diff
@@ -8,21 +8,146 @@
 
 #include "xpti/xpti_trace_framework.h"
 
+#include <dlfcn.h>
+#include <iostream>
+#include <mutex>
+#include <string>
 #include <sycl/detail/spinlock.hpp>
 
 sycl::detail::SpinLock GlobalLock;
 
 bool HasZEPrinter = false;
-bool HasCUPrinter = false;
-bool HasPIPrinter = false;
-bool HasSYCLPrinter = false;
 
-void zePrintersInit();
-void zePrintersFinish();
+std::string getCurrentDSODir() {
+  auto CurrentFunc = reinterpret_cast<const void *>(&getCurrentDSODir);
+  Dl_info Info;
+  int RetCode = dladdr(CurrentFunc, &Info);
+  if (0 == RetCode) {
+    // This actually indicates an error
+    return "";
+  }
+
+  auto Path = std::string(Info.dli_fname);
+  auto LastSlashPos = Path.find_last_of('/');
+
+  return Path.substr(0, LastSlashPos);
+}
```

Comment:
Is this meant to work on Windows? If so, does it?

Reply:
We should probably try to re-use `sycl/detail/os_util.cpp`  for this

Reply:
no, it is not intended to work on Windows. sycl-trace originally is not adapted for Windows so I didn't invest into it

Reply:
Fair! Would @npmiller's suggestion to use `sycl/detail/os_util.cpp` work though? Would mean a smaller maintenance surface and less work to make it work on Windows if need be in the future.

Reply:
it is internal file of sycl RT. Not sure if referencing it in separated tool is a good solution. But if we are talking about copy of file - no benefits for now.

Reply:
I was hoping maybe the tracer was able to share some files, but if that isn't the case then the current solution is probably better.